ore: Support uploading images to  IBMCloud

Prashanth684 opened the pull request:
One of the first steps in order to support Power Virtual server as a platform (https://github.com/coreos/fedora-coreos-tracker/issues/817) is the ability to upload images to the IBMCloud. The steps to upload to IBMCloud would be:

1. Make sure you have a valid account and create an API Key for authentication: https://cloud.ibm.com/docs/account?topic=account-userapikey
2. use the ore command to initialize the cloud object storage and storage bucket necessary to upload through `ore ibmcloud initialize`. The api key can be provided as a command line parameter or as a file. A resource group also needs to be provided based on what resource group the account belongs to: https://cloud.ibm.com/docs/account?topic=account-rgs
3. upload an image using the `ore ibmcloud upload` command. The cloud-object-storage and the bucket associated with it need to be specified if defaults are not used.

bgilbert reviewed (commented):
Please update `mantle/README.md` and `mantle/platforms.md`, starting with the information from the commit message.  Thanks!

(I didn't review the code, just happened to notice one thing)

Prashanth684 commented:
> Please update `mantle/README.md` and `mantle/platforms.md`, starting with the information from the commit message. Thanks!
> 
> (I didn't review the code, just happened to notice one thing)

updated the READMEs. i have updated them from the perspective of just supporting image uploads as we have to figure out the other stuff for actual VM creation which would happen later.

Prashanth684 commented on mantle/cmd/ore/ibmcloud/ibmcloud.go:
```diff
@@ -0,0 +1,124 @@
+var (
+	plog = capnslog.NewPackageLogger("github.com/coreos/mantle", "ore/ibmcloud")
+
+	IBMCLOUD = &cobra.Command{
```
sure i can change this. i was just being consistent with what other platforms did.

cgwalters reviewed (approved):
LGTM!

rpsene reviewed (commented):
/lgtm

clnperez commented on mantle/README.md:
````diff
@@ -350,6 +350,30 @@ for more information about the `.boto` file.
+```
+Once logged in, an api key can be created:
+```
+ $ ibmcloud iam api-key-create other_key --file ~/.ibmcloud/apikey.json --output json
````
just to cover all bases, and maybe @rpsene or @bpradipt can explain, but when using my version of `ibmcloud` (ibmcloud version 1.5.1+7684ebe-2021-04-27T18:17:58+00:00) -- the dir I end up with after running `ibmcloud login -sso` is  `~/.bluemix/`.  If I run this command, I get the error that the dir doesn't exist. 
```
ibmcloud iam api-key-create mykey --file ~/.ibmcloud/apikey.json --output json
FAILED
open /home/christy/.ibmcloud/apikey.json: no such file or directory
```

clnperez commented on mantle/cmd/ore/ibmcloud/ibmcloud.go:
```diff
@@ -0,0 +1,124 @@
+				os.Exit(1)
+			}
+			homedir := currentUser.HomeDir
+			credentialsFile = filepath.Join(homedir, ".ibmcloud/apikey.json")
```
depending on the answer from my earlier comment, this may need to change to ~/.bluemix. But again, maybe my env is borked. :D

Prashanth684 replied:
yup..will change...thanks a lot for this catch!

clnperez commented:
I have one question (see comments inline) about the location of the cred files (~/.ibmcloud v ~/.bluemix)

bpradipt commented on mantle/cmd/ore/ibmcloud/upload.go:
```diff
@@ -0,0 +1,118 @@
+		Example: `  ore ibmcloud upload --region=us-east \
+	  --cloud-object-storage=coreos-dev-image-ibmcloud \
+	  --bucket=coreos-dev-image-ibmcloud-us-east
+	  --file="/home/.../coreos_production_qcow_image.qcow2"`,
```
nit: the example can point to `raw` instead of `qcow2`

bpradipt replied:
/lgtm

Prashanth684 replied:
so the example cited here is for IBMCloud (x86) for which the default is qcow. the raw image is related to powervs which is a specific case so i didn't put that here.

Prashanth684 replied:
fixed the typo from raw -> qcow

Prashanth684 replied:
when we add in support for building OVAs , i will update the documentation here too.

bpradipt commented:
> I have one question (see comments inline) about the location of the cred files (~/.ibmcloud v ~/.bluemix)

@clnperez ibmcloud CLI creates `.bluemix` folder to keep local data :-)

Prashanth684 commented:
> @Prashanth684 also, I see some mention of a qcow2 variant, and that the source image format supported is raw. Is that all accurate?

so currently the examples i have cited is for the existing IBMCloud which produces x86 qcow images. The next step is to add support for building OVAs for PowerVS. Along with that i will also change the documentation here to reflect that it supports OVAs too. I was planning to take it one step at a time :)

Prashanth684 commented:
@bgilbert any objections to getting this merged?

bgilbert commented:
I haven't reviewed it, but no objections from my end.
